src/Illuminate/Support/Lottery.php

```diff
@@ -0,0 +1,140 @@
+<?php
+
+namespace Illuminate\Support;
+
+class Lottery
+{
+    /**
+     * The number of expected wins.
+     *
+     * @var int
+     */
+    protected $chances;
+
+    /**
+     * The number of potential opportunities to win.
+     *
+     * @var int
+     */
+    protected $outOf;
+
+    /**
+     * The winning callback.
+     *
+     * @var null|callable
+     */
+    protected $winner;
+
+    /**
+     * The losing callback.
+     *
+     * @var null|callable
+     */
+    protected $loser;
+
+    /**
+     * Create a new Lottery instance.
+     *
+     * @param  int  $chances
+     * @param  int  $outOf
+     */
+    public function __construct($chances, $outOf)
+    {
+        $this->chances = $chances;
+
+        $this->outOf = $outOf;
+    }
+
+    /**
+     * Create a new Lottery instance.
+     *
+     * @param  int  $chances
+     * @param  int  $outOf
+     * @return static
+     */
+    public static function odds($chances, $outOf)
+    {
+        return new static($chances, $outOf);
+    }
+
+    /**
+     * Set the winner callback.
+     *
+     * @param  callable  $callback
+     * @return $this
+     */
+    public function winner($callback)
+    {
+        $this->winner = $callback;
+
+        return $this;
+    }
+
+    /**
+     * Set the loser callback.
+     *
+     * @param  callable  $callback
+     * @return $this
+     */
+    public function loser($callback)
+    {
+        $this->loser = $callback;
+
+        return $this;
+    }
+
+    /**
+     * Run the lottery.
+     *
+     * @param  mixed  ...$args
+     * @return mixed
+     */
+    public function __invoke(...$args)
+    {
+        return $this->runCallback(...$args);
+    }
+
+    /**
+     * Run the lottery.
+     *
+     * @param  null|int  $times
+     * @return mixed
+     */
+    public function choose($times = null)
+    {
+        if ($times === null) {
```

Comment:
```suggestion
     * @param int  $times
     * @return mixed
     */
    public function choose($times = 1)
    {
        if ($times === 1) {
```

A great feature, but I think the default here could still be a numeric value, we are executing the call back one time instead of null times per default.

Reply:
Hey @abenerd thanks for the review!

This is intentional. If you don't pass an integer you get a single result, however if you pass an integer you will always get an `array` of results.


```php
$result = Lottery::odds(1, 1)->winner(fn () => 'winner')->choose(5);
assert(is_array($result));

$result = Lottery::odds(1, 1)->winner(fn () => 'winner')->choose(1);
assert(is_array($result));

$result = Lottery::odds(1, 1)->winner(fn () => 'winner')->choose();
assert(is_string($result));
```

This functionality matches the functionality seen in the `Arr::random($array, $count)` method.